synapse/cmds/cron.py

```diff
@@ -0,0 +1,598 @@
+import time
+import calendar
+import datetime
+import functools
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+import synapse.lib.cli as s_cli
+import synapse.lib.cmd as s_cmd
+import synapse.lib.time as s_time
+import synapse.lib.syntax as s_syntax
+
+StatHelp = '''
+Gives detailed information about a single cron job.
+
+Syntax:
+    cron stat <iden prefix>
+
+Notes:
+    Any prefix that matches exactly one valid cron job iden is accepted.
+'''
+
+DelHelp = '''
+Deletes a single cron job.
+
+Syntax:
+    cron del <iden prefix>
+
+Notes:
+    Any prefix that matches exactly one valid cron job iden is accepted.
+'''
+
+ListHelp = '''
+List existing cron jobs in a cortex.
+
+Syntax:
+    cron list
+
+Example:
+    cli> cron list
+    user       iden       recurs? now? # start last start       last end         query
+    <None>     4ad2218a.. N       N          1 2018-12-14T15:53 2018-12-14T15:53 #foo
+    <None>     f6b6aebd.. Y       N          3 2018-12-14T16:25 2018-12-14T16:25 #foo
+'''
+
+ModHelp = '''
+Changes an existing cron job's query.
+
+Syntax:
+    cron mod <iden prefix> <new query>
+
+Notes:
+    Any prefix that matches exactly one valid cron iden is accepted.
+'''
+
+AddHelp = '''
+Add a recurring cron job to a cortex.
+
+Syntax:
+    cron add [optional arguments] {query}
+
+    --minute, -M int[,int...][=]
+    --hour, -H
+    --day, -d
+    --month, -m
+    --year, -y
+
+Notes:
+    All times are interpreted as UTC.
+
+    All arguments are interpreted as the job period, unless the value ends in
+    an equals sign, in which case the argument is interpreted as the recurrence
+    period.  Only one recurrence period parameter may be specified.
```

Comment:
Is this an existing convention or a new one we are creating?

Reply:
A new one. The cron style uses asterisks, but that's problematic for command-ilne stuff. The alternative is to mark the recurring parameter.